### PR TITLE
Adjust `txt`, `csv` and `json` data sources to use the same exec attributes

### PR DESCRIPTION
**Pull request by traut**
## What was changed

- `txt`, `csv` and `json` data sources were adjusted to support both `path` and `glob` parameters
- the unit tests were updated

Fixes https://github.com/blackstork-io/fabric/issues/176

### CODE REVIEW

File: internal/testtools/diag.go [hunk 1]
```diff
@@ -57,14 +110,21 @@ func CompareDiags[D diagnostics.Generic](t *testing.T, fm map[string]*hcl.File,
...
+		var b strings.Builder
+		b.WriteString("Actual diagnostics:\n")
+		for _, diag := range diags {
+			b.WriteString("\n")
+			b.WriteString(fmt.Sprintf("[Severity]: %s\n", severityToString(diag.Severity)))
+			b.WriteString(fmt.Sprintf("[Summary]: %s\n", diag.Summary))
+			b.WriteString(fmt.Sprintf("[Details]: %s\n", diag.Detail))
+			b.WriteString("\n")
+		}
+		t.Fatal(b.String())
```
Comment (traut) [inline internal/testtools/diag.go:123]: @Andrew-Morozko I found this output easier to read when debugging tests, but maybe there are edge cases of `compareDiags` usage I've missed!

Comment (Andrew-Morozko) [reply internal/testtools/diag.go:123]: Yep, I see a problem with this approach: in your case the order of the asserts must match the order of diagnostics. I made this assert system so it doesn't care about the order of the diagnostics coming in. 

I expect that the diagnostic order will be most of the time the same, but parallel code might sometimes emmit them in a different order, creating a semi-broken test. And besideds, I don't think that *order* of the diagnostics shouldn't matter, only that they match the given diags.

Regarding improvements to the output, I've repeated your work at the exact same time 😅. If you want to change something else - I thing #180 could be the place

Comment (traut) [reply internal/testtools/diag.go:123]: go ahead, Andrew! I'll rebase this on #180

**Review by Andrew-Morozko — CHANGES_REQUESTED** [review]
Sorry for the conflicts here, I'll pick something smaller to work on next time 😅

**Comment by traut** [comment]
closing in favor of https://github.com/blackstork-io/fabric/pull/186
